internal/femto/authentication.go

```diff
@@ -0,0 +1,50 @@
+package femto
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+)
+
+var (
+	NotAuthenticatedError   = errors.New("User does not have a valid authentication token")
+	InvalidCredientalsError = errors.New("Invalid credientals for creating an authentication token")
+)
+
+const (
+	AuthCookieField = "auth_cookie"
+)
+
+type AuthToken = string
+
+type Authenticator[AuthCredientals any] interface {
+	CreateToken(AuthCredientals) (AuthToken, error)
+	RevokeToken(AuthToken) error
+	CheckToken(AuthToken) bool
+}
+
+func AuthWrapGet[A any, T any](auth Authenticator[A], handle GetFunc[T]) GetFunc[T] {
+	return func(rr *http.Request, ll *slog.Logger) (T, error) {
+		f := AuthWrapPost(auth, func(zero struct{}, r *http.Request, l *slog.Logger) (T, error) {
+			return handle(r, l)
+		})
+		return f(struct{}{}, rr, ll)
+	}
+}
+
+func AuthWrapPost[A any, T any, R any](auth Authenticator[A], handle PostFunc[T, R]) PostFunc[T, R] {
+	return func(data T, r *http.Request, l *slog.Logger) (R, error) {
+		var zero R
```

Comment:
This `R` type is concerning. From what I've gathered this has been added due to the authentication token providing endpoint being a POST endpoint. This really shouldn't be the case.

POST semantics are `a -> ()`. GET semantics are `a -> b`. That is to say, POST is side-effecting, GET is not, at least from the perspective of the user. This is a useful lie, as when we try to get the auth token from the server, we need to alter state in such a way that the auth token is remembered. What is important, is the state transition **is not** a function of the data provided by the user. In the case of UUID generation, it's a function of random entropy. 

For the user, this can be modelled and reasoned by a some form of pre-generated tokens, or whatever else is useful. However the user chooses to model these semantics, the user need not worry about needing to reason about state transitions given they're entirely local to the groundstation. From the perspective of the user, I would expect the API to be GET.

This prevents the heavy abstraction leakage we see here. Almost no other POST endpoints will need to return any sort of data, and just act as side-effecting endpoints. We will have an absurd amount of endpoints that need to explicitly say that they don't do any sort of pure computation, rather than this being implicit in the type. We can already see that in the changes in the groundstation post endpoints. 

We'd significantly reduce the size of this PR by setting the auth endpoint to be GET.